TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -77,11 +81,31 @@ class MatchEngineUnit : public MatchEngineUnitBase<VMProjType> {
 
   good__ = false;
 
+  iuse_ = 0;
+  bool usefirstMinus = projbuffer_.getUseFirstMinus();
+  bool usesecondMinus = projbuffer_.getUseSecondMinus();
+  bool usefirstPlus = projbuffer_.getUseFirstPlus();
+  bool usesecondPlus = projbuffer_.getUseSecondPlus();
+#pragma HLS ARRAY_PARTITION variable=use_ dim=0 complete
+  clearuse: for(int iuse = 0; iuse < kNuse; iuse++) {
+#pragma HLS unroll
+    use_[iuse] = 0;
+  } 
+  nuse_ = 0;
+  if(usefirstMinus) use_[nuse_++] = 0;//(0, 0);
+  else stubmask_[0] = 0;
+  if(usesecondMinus) use_[nuse_++] = 2;//(1, 0);
+  else stubmask_[1] = 0;
+  if(usefirstPlus) use_[nuse_++] = 1;//(0, 1);
+  else stubmask_[2] = 0;
+  if(usesecondPlus) use_[nuse_++] = 3;//(1, 1);
+  else stubmask_[3] = 0;
+
 }
 
 
 
-inline void step(const VMStubMECM<VMSMEType> stubmem[4][1024]) {
+inline void step(const VMStubMECM<VMSMEType> stubmem[4][1<<(kNbitsrzbinMP+kNbitsphibinMP+4)]) {
```

Comment:
Why is this changed to 4096 entries?

Reply:
With the extra bit required for the disks, this memory grew. I'll push the properly templated fix soon.

Reply:
But it should be 2048 for the disks?

Reply:
With this new change
https://github.com/cms-L1TK/firmware-hls/blob/2662e01f7ecf90a69e6f98e303403cffa5f8c830/TrackletAlgorithm/MatchProcessor.h#L1177-L1178
only the disk R/Z is changed to 4, so everything should be the correct length now.